src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,205 @@
+            InvalidInputException {
+        String output = "";
+        if (command.equals("bye")) { // exit
+                throw new DukeTerminateException();
+            } else if (command.equals("list")) {
+                output = taskList.list();
+            } else if (command.split(" ")[0].equals("find")) { //find
+                if (command.split(" ").length < 2) {
+                    throw new InsufficientInputException();
+                }
+                String key = command.split(" ")[1].trim();
+                output = taskList.find(key);
+            } else if (command.split(" ")[0].equals("done")) { // done
+                if (command.split(" ").length < 2) {
+                    throw new InsufficientInputException();
+                }
+                int idx = Integer.parseInt(command.split(" ")[1]) - 1;
+                output = taskList.done(idx);
+            } else if (command.split(" ")[0].equals("delete")) { // delete
+                if (command.split(" ").length < 2) {
+                    throw new InsufficientInputException();
+                }
+                int idx = Integer.parseInt(command.split(" ")[1]) - 1;
+                output = taskList.delete(idx);
+            } else if (command.split(" ")[0].equals("todo")) { // add todo
+                if (command.split(" ").length < 2) {
+                    throw new InvalidTodoException();
+                }
+                command = command.substring(command.split(" ")[0].length() + 1, command.length());
+                Task task = new Task(command);
+                output = taskList.addTask(task);
+            } else if (command.split(" ")[0].equals("deadline")) { // add deadline
+                command = command.substring(command.split(" ")[0].length() + 1, command.length()).trim();
+                if (command.split("/by").length == 0) {
+                    throw new InvalidDeadlineException("OOPS! The description and time is missing. " +
+                            "Format: description /by time");
+                } else if (command.split("/by").length == 1) {
+                    if (command.split("/by")[0].equals("")) {
+                        throw new InvalidEventException("OOPS! The description and time is missing. " +
+                                "Format: description /by time");
+                    } else {
+                        throw new InvalidEventException("OOPS! The time is missing. " +
+                                "Format: description /by time");
+                    }
+                } else if (command.split("/by").length == 2) {
+                    if (command.split("/by")[0].equals("")) {
+                        throw new InvalidDeadlineException("OOPS! The description is missing. " +
+                                "Format: description /by time");
+                    }
+                }
+                String name = command.split(" /by")[0].trim();
```

Comment:
I noticed a lot of your code for split() is repeated multiple times, would it be easier to extract that out as a function?

Reply:
If you are repeating code, it is (usually) bad practice. Think of how you can abstract them into separate functions!